docs/src/ci.md

```diff
@@ -0,0 +1,143 @@
+
+## running locally
+
+If you want to run the script locally, you can set custom URLs via environment variables. For example:
+
+```bash
+CI_UNIT_PKG_URL_QEDbase="www.github.com/User/QEDbase#0e1593b" CI_UNIT_PKG_URL_QEDfields="www.github.com/User/QEDfields#60324ad" julia --project=/path/to/QED/repo SetupDevEnv.jl`
+```
+
+# Integration Tests for CI Develops
+
+In this section, we explain how the integration tests are created and executed. It is not mandatory to read the section if you only want to use the CI. The following figure shows the stages of the CI pipeline:
+
+![detailed CI pipeline of the integration tests](integration_jobs_pipeline.svg)
+
+All of the following stages are executed in the CI pipeline of the sub-package, where the code is modified. This means also, that GitLab CI automatically checks out the repository with the changes of the pull request and provides it in the CI job. We name the package `orig` for easier understanding of the documentation.
+
+- **Stage: Integration Tests**: Execute the unit tests of `orig` via `Pkg.test()`.
+- **Stage: Generate integration Tests**: Download the `integTestGen.jl` script from the `QED.jl` package via `git clone`. The script is executed with the name of the subpackage `orig`. The script traverses the dependency tree of the `QED.jl` and it is searching for sub-packages which has a dependency of package `orig`. For each package with the dependency of package `orig`, the generator generates a job yaml. By default, it uses the upstream repository and development branch of the package tested in the integration test. The repository and commit can be changed via the environment parameter in the git commit message. The `integTestGen.jl` uses the `Package.toml` of the `QED.jl` package because all sub-packages are direct and indirect dependencies of it.
+- **Stage: Run Integration Tests**: This stage uses the generated job yaml to create and run new test jobs. It uses the [GitLab CI child pipeline](https://about.gitlab.com/blog/2020/04/24/parent-child-pipelines/#dynamically-generating-pipelines) mechanism.
+- **Stage: Integration Tests of Sub package N**: Each job clones the repository of the subpackage. After the clone, it uses the Julia function `Pkg.develop(path="$CI_package_DIR")` to replace the dependency to the package `orig` with the modified version of the Pull Request and execute the tests of the subpackage via `Pkg.test()`.
```

Comment:
Sub package, subpackage, or sub-package? It should be the same everywhere.

Again, I don't get who depends on whom.

Also:
```suggestion
- **Stage: Integration Tests of Sub package N**: Each job clones the repository of the subpackage. After the clone, it uses the Julia function `Pkg.develop(path="$CI_package_DIR")` to replace the dependency to the package `orig` with the modified version of the pull request and execute the tests of the subpackage via `Pkg.test()`.
```